### ISSUE TITLE
grdtrack -C does not honor distances in degrees

### ISSUE THREAD

**Issue by PaulWessel** [GenericMappingTools/gmt#5255]
**Description of the problem**

For geographic grids, the values passed via **-C** is assumed to be in distance units (e.g., km, miles, etc) but if the user gives them in degrees we get junk since there is an assumption that the line spacing and intervals can be converted from these units to degrees when in fact this should not happen when degrees was given in the first place.  Offending line in _gmtsupport_crosstracks_spherical_:

`across_ds_radians = D2R * (cross_half_width / GMT->current.proj.DIST_M_PR_DEG) / n_half_cross;	/* Angular change from point to point */`

Obviously, the meter to degree part is wrong in this case.  Workaround until the bug fix is to give the equivalent spacing in km.

**Comment by joa-quim** [comment]
One other think with the distances is that we seem to have a kind of grid vs pix registration issue. Using ``-C10k/2`` computes distances that are approximately 2.5 km instead of the the requested 2 km. From a own example
```
> Cross profile number -L0-331 at  -13.280/043.700 az=090.0
-13.3421964932  43.6999831383   -5.01067470707
-13.311098251   43.6999957846   -2.50533735518
-13.28  43.7    0
-13.248901749   43.6999957846   2.50533735518
-13.2178035068  43.6999831383   5.01067470707
```

**Comment by PaulWessel** [comment]
Well, I think this is more of a mismatch with your input selection/not enough documentation.  We compute:

	n_half_cross = irint (cross_length / across_ds);	/* Half-width of points in a cross profile */

since you have 5 / 2 it is not an integer and we go to the nearest one.  5 is the half-length of 10 and we want that to divide nicely into across_ds.  I think I will add a warning when that happens which would have tipped you off.

**Comment by joa-quim** [comment]
Ok, with the warning but I think the requested *inc* should be respected, even if it's silly.

**Comment by PaulWessel** [comment]
Will have to see how easy that is.  It was created with the assumption we wanted to sample the point defined by the main line, but now you want to skip that by being 0.01 on one side and 0.01 on the other side...

**Comment by joa-quim** [comment]
Well, I just think that if we say 
*length sets the full length of each cross-profile, while ds is the sampling spacing along each cross-profile.*
we would better respect it.

**Comment by PaulWessel** [comment]
Also, I suspect you actually want Cartesian behavior in your crossprofiles, in that all your profiles should be spaced by 0.02 in longitude.  But that is not going to happen in a spherical world since 0.02d is great circle distances in degrees, not longitude increments. So depends on what you are trying to do here.  Maybe treating the grid as Cartesian is what you want, via -fc?

**Comment by PaulWessel** [comment]
Have implemented the case where we honor the increments.

**Comment by joa-quim** [comment]
Yes, it was a cartesian behavior. Small error since I only wanted to stack a couple columns. It was also pure laziness to not implement a *nanmean* in Julia, which I ended up doing.

**Comment by PaulWessel** [comment]
Was fixed by #5257.
